accounts-db/src/accounts_db.rs

```diff
@@ -8981,16 +8981,17 @@ impl AccountsDb {
         self.accounts_index.scan(
             pubkeys.iter(),
             |_pubkey, slots_refs, _entry| {
-                if let Some((slot_list, ref_count)) = slots_refs {
-                    if slot_list.len() == 1 && ref_count == 1 {
-                        if let Some((slot_alive, acct_info)) = slot_list.first() {
-                            if acct_info.is_zero_lamport() && !acct_info.is_cached() {
-                                count += 1;
-                                self.zero_lamport_single_ref_found(*slot_alive, acct_info.offset());
-                            }
-                        }
+                let (slot_list, ref_count) = slots_refs.unwrap();
```

Comment:
why move these to `unwrap` instead of `if let Some(...`

Reply:
@brooksprumo can slots_ref be none at startup?

Reply:
i can understand that. just seems maybe unnecessary to add a potentially dangers `unwrap` when the existing code was tolerant. And, I'd probably like this little set of code shared. I think we do it 2-3 times.

Reply:
i imagine it is ok as it is now.

Reply:
i did try to find the comment i imagined you made, @brooksprumo I imagined requesting this but it didn't show up when i was looking at the code.